pkg/providers/tinkerbell/hardware/json.go

```diff
@@ -14,17 +16,15 @@ import (
 // ErrTinkebellHardwareJsonRepeatWrites occurs when a TinkerbellHardwareJson receives multiple calls to Write().
 var ErrTinkebellHardwareJsonRepeatWrites = errors.New("TinkerbellHardwareJson can only be written to once")
 
-// TinkerbellHardwareJsonFactory creates new TinkerbellHardwareJson instances. The path used by the
-// TinkerbellHardwareJson is available as the second return parameter.
+// TinkerbellHardwareJsonFactory creates new TinkerbellHardwareJson instances.
 type TinkerbellHardwareJsonFactory interface {
-	Create(name string) (*TinkerbellHardwareJson, string, error)
```

Comment:
I changed the approach to retrieving all the data we want to register by instrumenting a journal when we write the output. See [`RecordingTinkerbellHardwareJsonFactory()`](https://github.com/aws/eks-anywhere/pull/1507/files#diff-17e36707f7afa6b3bdcdd809870cb9d2456e9b703b4cfc26a73fa9b90279d9ccR154) for details.